src/org/opendatakit/aggregate/parser/BaseFormParserForJavaRosa.java

```diff
@@ -311,10 +312,10 @@ private static synchronized final XFormParserWithBindEnhancements parseFormDefin
   /**
    * The ODK Id that uniquely identifies the form
    */
-  protected final FormDef rootJavaRosaFormDef;
+  protected transient final FormDef rootJavaRosaFormDef;
```

Comment:
I make all all non-serializable class transient. I think this is why you're getting the NPE in your test because it reference some elements that are not restored from the deserialization process.